frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1610,6 +1610,42 @@ function ReaderHighlight:getHighlightBookmarkItem()
     end
 end
 
+function ReaderHighlight:_insertHighlight(page, highlight)
+    local position
+    if self.view.highlight.saved[page] == nil or #self.view.highlight.saved[page] == 0 then
+        position = 1
+    else
+        local page_highlights = self.view.highlight.saved[page]
+        local highlight_beginning = highlight.pos0
+        if type(highlight_beginning) ~= "string" then
+            highlight_beginning.page = page
```

Comment:
It is needed for pdf only, so can be done within the following if-else-condition.

Reply:
Done.

Reply:
Done.